std/datetime/interval.d

```diff
@@ -1536,47 +1537,38 @@ public:
         return range;
     }
 
-
-    /+
-        Converts this interval to a string.
-      +/
-    // Due to bug http://d.puremagic.com/issues/show_bug.cgi?id=3715 , we can't
-    // have versions of toString() with extra modifiers, so we define one version
-    // with modifiers and one without.
-    string toString()
-    {
-        return _toStringImpl();
-    }
-
-
     /++
         Converts this interval to a string.
+        Params:
+            w = A `char` accepting
+            $(REF_ALTTEXT output range, isOutputRange, std, range, primitives)
+        Returns:
+            A `string` when not using an output range; `void` otherwise.
       +/
-    // Due to bug http://d.puremagic.com/issues/show_bug.cgi?id=3715 , we can't
-    // have versions of toString() with extra modifiers, so we define one version
-    // with modifiers and one without.
-    string toString() const nothrow
-    {
-        return _toStringImpl();
-    }
-
-
-private:
-
-    /+
-        Since we have two versions of toString, we have _toStringImpl
-        so that they can share implementations.
-      +/
-    string _toStringImpl() const nothrow
+    string toString() const @safe nothrow
     {
-        import std.format : format;
+        import std.array : appender;
+        auto app = appender!string();
         try
-            return format("[%s - %s)", _begin, _end);
+            toString(app);
         catch (Exception e)
-            assert(0, "format() threw.");
+            assert(0, "toString() threw.");
+        return app.data;
```

Comment:
I think I have now seen this for the 20th time. Couldn't we add a toString function in std.format or somewhere else that is defined exactly when there is no toString but a output range toString?